sdk/communication/Azure.Communication.PhoneNumbers/src/SipRouting/SipConfiguration.cs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text.Json;
+using Azure.Core;
+
+namespace Azure.Communication.PhoneNumbers.SipRouting
+{
+    [CodeGenSuppress("SipConfiguration")]
+    internal partial class SipConfiguration
+    {
+        /// <summary>
+        /// SIP trunks for routing calls.
+        /// Map key is trunk&apos;s FQDN (1-249 characters).
+        /// </summary>
+        internal IReadOnlyDictionary<string, SipTrunk> Trunks { get; }
+
+        /// <summary> Trunk routes for routing calls. </summary>
+        internal IReadOnlyList<SipTrunkRoute> Routes { get; }
+
+        internal SipConfiguration(IDictionary<string, SipTrunk> trunks)
+        {
+            Trunks = new ReadOnlyDictionary<string, SipTrunk>(trunks);
+        }
+
+        internal SipConfiguration(IEnumerable<SipTrunkRoute> routes)
+        {
+            Routes = routes.ToList().AsReadOnly();
+        }
+
+        // <summary> Initializes a new instance of SipConfiguration. </summary>
+        /// <param name="writer"></param>
```

Comment:
I'd add a summary for the parameter as well, explaining what it is and/or it's used for.

Reply:
Summary added.